lib/erblint-github/linters/github/accessibility/details_has_summary.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require_relative "../../custom_helpers"
+
+module ERBLint
+  module Linters
+    module GitHub
+      module Accessibility
+        class DetailsHasSummary < Linter
+          include ERBLint::Linters::CustomHelpers
+          include LinterRegistry
+
+          MESSAGE = "<details> elements need to have explict <summary> elements"
+
+          def run(processed_source)
+            tags(processed_source).each do |tag|
+              next if tag.closing?
+              next unless tag.name == "details"
+
+              # Find the details element index in the AST
+              index = processed_source.ast.to_a.find_index(tag.node)
+
+              # Get the next element in the AST
```

Comment:
Do you know if there's any reason to support `<summary>` not being the immediate next element? I don't know it's valid anyways but worth verifying.

```.html.erb
<details>
  <p> some text </p>
  <summary>Toggle</summary>
</details>

Reply:
There isn't an easy way for `getting the children of a element` with erblint since we only have access to the tag nodes without the actual HTML tree structure. 

One approach we've taken is keeping track of the opening/closing tag with variables. Here is an examples where we do this in dotcom:

- [SvgHasAccessibleTitle](https://github.com/github/github/blob/master/.erb-linters/accessibility/a11y_svg_has_accessible_text.rb#L15)

This approach might make sense if we wanted to allow `<summary>` that isn't a first child, or if we wanted to give a specific warning message for `<summary>` that isn't first child.

Reply:
> Do you know if there's any reason to support `<summary>` not being the immediate next element? I don't know it's valid anyways but worth verifying.
> 
> ```
> <details>
>   <p> some text </p>
>   <summary>Toggle</summary>
> </details>
> ```

Hmm, I thought it _needed_ to be the first element but that doesn't seem to be true. I wonder if it's worth supporting or if we should also validate that `<summary>` is the first element for consistency 🤔 

I'll change this to allow `<summary>` to be anywhere within `<details>` for now.